snapshot.py

```diff
@@ -81,34 +85,47 @@ def read_stats(args):
        (args.host != '' and args.user != 'root'):
         LOGGER.warning("If not running as root you may not see all info.")
 
-    optional_port = ''
-    if args.ssh_port:
-        optional_port = '-p %d' % args.ssh_port
-
+    cmd = "bash -c \"%s\"" % (cmd % (pids, pids))
     if args.host == '':
         LOGGER.info('Loading local procfs files')
-        cmd = "bash -c \"%s\"" % (cmd % (pids, pids))
+        p = Popen(cmd, shell=True, bufsize=-1, stdout=PIPE, stderr=PIPE)
+        stdout = p.stdout
+        stderr = p.stderr
     elif args.host != '':
-        ssh = (
-            "ssh %s %s@%s"
-            " -o UserKnownHostsFile=/dev/null"
-            " -o StrictHostKeyChecking=no"
-            " -o LogLevel=error"
-            % (optional_port, args.user, args.host)
-        )
+        ssh = paramiko.SSHClient()
+        # INSECURE, but reasonable for STBs
+        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
         if args.password:
-            ssh = "sshpass -p %s %s" % (args.password, ssh)
+            ssh.connect(args.host, port=args.ssh_port, username=args.user, password=args.password)
         else:
-            ssh = "%s -o PasswordAuthentication=no" % ssh
-
-        cmd = """%s "%s" """ % (ssh, cmd % (pids, pids))
+            if not args.key:
+                args.key = os.path.expanduser('~/.ssh/id_rsa')
+            mykey = paramiko.RSAKey.from_private_key_file(args.key)
+            ssh.connect(args.host, username=args.user, pkey=mykey)
+
+        p = ssh.get_transport().open_session()
+        p.exec_command(cmd)
```

Comment:
I implemented it this way so that I would have use of p.recv_exit_status for error checking. I'll revisit that. There may be another way of checking for exit/error, or it may not be useful at all.